app/api/internal/stats.rb

```diff
@@ -0,0 +1,23 @@
+module ThreeScale
+  module Backend
+    module API
+      internal_api '/services/:service_id/stats' do
+        before do
+          respond_with_404('service not found') unless Service.exists?(params[:service_id])
+        end
+
+        delete '' do |service_id|
+          begin
+            delete_stats_job_attrs = api_params Stats::DeleteJobDef
+            delete_stats_job_attrs[:service_id] = service_id
+            Stats::DeleteJobDef.new(delete_stats_job_attrs).run_async
```

Comment:
I think this is the only line that can raise. If so, it should be the only one in `begin`.

Reply:
:+1: